### ISSUE TITLE
Do I need to split my edge_index in GAE?

### ISSUE THREAD

**Issue by levulinh** [rusty1s/pytorch_geometric#2057]
## ❓ Questions & Help

Hi, I am recently trying to use GAE. I want to fit the whole graph into GAE, unlike in the case of the example which splits the edge_index into pos_edge_index and neg_edge_index. Here are my questions:
1. Since I need to get the latent embeddings of all nodes, can I feed edge_index to the encoder instead of train_pos_edge_index?
2. I don't really get the meaning of positive and negative edges, can you explain what are they and how are they splitted from the original set of edges? If I want to get AUC, AP on evaluation, can I just split my edge_index in halves and treat them like pos and neg edges, and then fit the whole graph into the test function?
3. Edit: I had a question about fitting many graphs into GAE, but then I search for similar issues more carefully and found out setting cache=False should do it.
Thank you very much in advanced!

**Comment by rusty1s** [comment]
1. You can simply pass `edge_index` to the `forward` method instead of `train_pos_edge_index`.
2. Positive edges correspond to the edges that exist in the data, while negative edges refer to edges that should not exist. To evaluate a link prediction model, it is necessary for the model to distinguish those (think of them as two different labels). We want our model to output high confidence for an edge in case the edge really exists, while we want low probability for edge existence in case we input a negative edge. You will need negative edges to evaluate/train your model. You can use `torch_geometric.utils.negative_sampling` to sample such edges.

**Comment by levulinh** [comment]
Thank you for quick reply. So for evaluating the model, I can feed my edge_index as positive edges and edges gotten from negative_sampling as negative edges right?

**Comment by rusty1s** [comment]
Yes, but keep in mind that you want to avoid data leakage going from training to testing. That's the reason we do not simply use `edge_index` during training, since this data already contains positive test edges.
